src/lib/hooks/use-user-content-anchor-links.ts

```diff
@@ -0,0 +1,103 @@
+import { useRouter } from 'next/router'
+import { useEffect } from 'react'
+
+/**
+ * When the #hash in the page URL changes, or a relevant anchor link is clicked,
+ * react accordingly by calling `handleUserContentHash`, which jumps to
+ * `user-content-` anchor links even if the hash in the URL or `href` does
+ * not explicitly contain the `user-content-` prefix.
+ *
+ * Note this is necessary for sanitized markdown content, as `id` attributes
+ * are automatically prefixed with "user-content-" by `rehype-sanitize`.
+ * Despite this `id` sanitization, we want users to share and use hash URLs
+ * without the `user-content-` prefix. On the same note, for our own convenience
+ * in implementation, we want to be able to create links to `#hashes` without
+ * worrying about the effects of sanitization.
+ *
+ * This solution is adapted for NextJS based on the `rehype-sanitize` example:
+ * https://github.com/rehypejs/rehype-sanitize/blob/main/readme.md#example-headings-dom-clobbering
+ * (If link breaks, search `scrollIntoView` on the `rehype-sanitize` readme).
+ */
+function useUserContentAnchorLinks() {
```

Comment:
Have tried to go heavy on the comments here, since dealing with anchor links in sanitized content isn't something we've done before in this repo. The TLDR is there's an example implementation at <https://github.com/rehypejs/rehype-sanitize/blob/main/readme.md#example-headings-dom-clobbering> which informed this work.